pcap4j-core/src/main/java/org/pcap4j/packet/IcmpV6MobilePrefixSolicitationPacket.java

```diff
@@ -0,0 +1,222 @@
+         * @return this Builder object for method chaining.
+         */
+        public Builder identifier(short identifier) {
+            this.identifier = identifier;
+            return this;
+        }
+
+        /**
+         * @param reserved reserved
+         * @return this Builder object for method chaining.
+         */
+        public Builder reserved(short reserved) {
+            this.reserved = reserved;
+            return this;
+        }
+
+        @Override
+        public IcmpV6MobilePrefixSolicitationPacket build() {
+            return new IcmpV6MobilePrefixSolicitationPacket(this);
+        }
+
+    }
+
+    public static final class IcmpV6MobilePrefixSolicitationHeader extends AbstractHeader {
+
+        /*
+         *  0                            15                              31
+         * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+         * |          Identifier           |            Reserved           |
+         * +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+         */
+
+        /**
+         *
+         */
+        private static final long serialVersionUID = -2991706817314703570L;
+
+        private static final int IDENTIFIER_OFFSET = 0;
+        private static final int IDENTIFIER_SIZE = SHORT_SIZE_IN_BYTES;
+        private static final int RESERVED_OFFSET = IDENTIFIER_OFFSET + IDENTIFIER_SIZE;
+        private static final int RESERVED_SIZE = SHORT_SIZE_IN_BYTES;
+        private static final int ICMPV6_MOBILE_PREFIX_SOLICITATION_HEADER_SIZE = RESERVED_OFFSET + RESERVED_SIZE;
+
+        private final short identifier;
+        private final short reserved;
+
+        private IcmpV6MobilePrefixSolicitationHeader(byte[] rawData, int offset, int length)
+                throws IllegalRawDataException {
+            if (length < ICMPV6_MOBILE_PREFIX_SOLICITATION_HEADER_SIZE) {
+                StringBuilder sb = new StringBuilder();
+                sb.append("The data is too short to build an ICMPv6 Mobile Prefix Solicitation Header(")
+                        .append(ICMPV6_MOBILE_PREFIX_SOLICITATION_HEADER_SIZE)
+                        .append(" bytes). data: ")
+                        .append(ByteArrays.toHexString(rawData, " "))
+                        .append(", offset: ")
+                        .append(offset)
+                        .append(", length: ")
+                        .append(length);
+                throw new IllegalRawDataException(sb.toString());
+            }
+            this.identifier = ByteArrays.getShort(rawData, IDENTIFIER_OFFSET + offset);
+            this.reserved = ByteArrays.getShort(rawData, RESERVED_OFFSET + offset);
+        }
+
+        private IcmpV6MobilePrefixSolicitationHeader(Builder builder) {
+            this.identifier = builder.identifier;
+            this.reserved = builder.reserved;
+        }
+
+        public short getIdentifier() {
+            return identifier;
+        }
+
```

Comment:
Please add a method that returns the identifier as int like below:

```java
public int getIdentifierAsInt() {
  return identifier & 0xFFFF;
}
```

Reply:
added.